cpp/osh.cc

```diff
@@ -17,7 +17,7 @@ using syntax_asdl::loc;
 namespace arith_parse {
 
 GcGlobal<tdop::ParserSpec> kArithSpec_ = {
-    {TypeTag::OtherClass, kZeroMask, HeapTag::Global, kIsGlobal}};
+    {kNotInPool, TypeTag::OtherClass, kZeroMask, HeapTag::Global, kIsGlobal}};
```

Comment:
Aside: Should we add change ObjHeader to be constructed via a constructor instead of aggregate init so we don't have to change all these places when changing the header?